public/hist_nd.pro

```diff
@@ -131,13 +133,25 @@
 ;
 ;##############################################################################
 
-function hist_nd,V,bs,MIN=mn,MAX=mx,NBINS=nbins,REVERSE_INDICES=ri
+function hist_nd,V,bs,MIN=mn,MAX=mx,NBINS=nbins,REVERSE_INDICES=ri,NAN=nan
+
+  nan=(n_elements(nan) ne 0) && keyword_set(nan)
+  if (~nan && ~array_equal(finite(V), 1)) then begin
+    message, 'NaN detected; setting NAN keyword', /continue
+    nan=1
+  endif
+
   s=size(V,/DIMENSIONS)
   if n_elements(s) ne 2 then message,'Input must be N (dimensions) x P (points)'
   if s[0] gt 8 then message, 'Only up to 8 dimensions allowed'
   
-  imx=max(V,DIMENSION=2,MIN=imn)
+  imx=max(V,DIMENSION=2,MIN=imn,NAN=nan)
   
+  if (nan && $
+      (array_equal(finite(imx), 0) || $
+       array_equal(finite(imn), 0))) then $
+    message, 'No finite values in at least one column', /continue
```

Comment:
Same thing - either quietly continue or throw an error.

Reply:
I'll remove for the PR.  I found it made debugging easier to know this.